### PR TITLE
Read latest version in downloads from release data

### PR DESCRIPTION
**Pull request by rollerozxa**
Simplifies things slightly during release, updating the release data in `_data/release.yml` will update the downloads page.

### CODE REVIEW

File: downloads.html [hunk 1]
```diff
@@ -80,13 +82,13 @@ <h2>Android</h2>
...
-            <a href="https://github.com/minetest/minetest/releases/download/5.9.1/minetest-5.9.1-armeabi-v7a.apk">32-bit</a>
+            <a href="https://github.com/minetest/minetest/releases/download/{{ latest_ver }}/minetest-{{ latest_ver }}-arm64-v8a.apk">64-bit</a> -
+            <a href="https://github.com/minetest/minetest/releases/download/{{ latest_ver }}/minetest-{{ latest_ver }}-armeabi-v7a.apk">32-bit</a>
           </li>
           <li>
             x86:
-            <a href="https://github.com/minetest/minetest/releases/download/5.9.1/minetest-5.9.1-x86_64.apk">64-bit</a> -
-            <a href="https://github.com/minetest/minetest/releases/download/5.9.1/minetest-5.9.1-x86.apk">32-bit</a>
+            <a href="https://github.com/minetest/minetest/releases/download/{{ latest_ver }}/minetest-{{ latest_ver }}-x86_64.apk">64-bit</a> -
+            <a href="https://github.com/minetest/minetest/releases/download/{{ latest_ver }}/minetest-{{ latest_ver }}-x86.apk">32-bit</a>
```
[the remark this answers is not in the document]
Comment (rollerozxa) [reply downloads.html:91]: Well the Android build is made in CI now so there shouldn't be any delay in getting them ready? Any coredev could then sign the APKs when making the release if you've got the keystore password which I assume would be the case.

File: _data/release.yml [hunk 2]
```diff
@@ -4,3 +4,5 @@ latest:
   date: "2024-09-15"
   protocol_version: 45
   url: "https://www.minetest.net/downloads/"
+  # Header for the latest changelog entry on the Wiki
+  changelog_header: "5.9.0_.E2.86.92_5.9.1"
```
[the remark this answers is not in the document]
Comment (rollerozxa) [reply _data/release.yml]: Yes that would work too.

File: downloads.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (sfan5) [reply downloads.html:91]: > which I assume would be the case

it's not.
also if we get the macos thing working we would have to wait for one specific coredev before we can publish a signed release.

[the remark this answers is not in the document]
Comment (rollerozxa) [reply downloads.html:91]: I feel that's an issue with the release process that would need some discussion among the core team to give more coredevs access to the keystore password so it doesn't need to fall on one or two coredevs who may be absent during release time, but fair enough.

[the remark this answers is not in the document]
Comment (sfan5) [reply downloads.html:91]: :shrug:

**Pull request closed without merging**